pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -127,6 +127,7 @@ func (a *AgentPXEFiles) PersistToFile(directory string) error {
 		return err
 	}
 	defer kernelReader.Close()
+	defer os.RemoveAll(a.tmpPath)
```

Comment:
Why not put this at the top of the function?

Reply:
It was mainly put there for readability, to be close to where it was used. I can move it if there is a preference to have it at the top.